src/components/ServiceDetails/AboutService/ServiceDemo/index.js

```diff
@@ -4,30 +4,68 @@ import { connect } from "react-redux";
 
 import ProgressBar from "../../../common/ProgressBar";
 import { useStyles } from "./styles";
-import ThirdPartyAIService from "./ThirdPartyAIService";
+import { serviceActions } from "../../../../Redux/actionCreators";
+import PurchaseToggler from "./PurchaseToggler";
 
 class ServiceDemo extends Component {
   state = {
     error: "error state message",
-    progressText: ["Configure", "Results"],
+    progressText: ["Purchase", "Configure", "Results"],
+    purchaseCompleted: false,
+  };
+
+  componentDidMount = () => {
+    this.fetchFreeCallsUsage();
+  };
+
+  fetchFreeCallsUsage = async () => {
+    const { service, fetchMeteringData, email } = this.props;
+    fetchMeteringData({
+      orgId: service.org_id,
+      serviceId: service.service_id,
+      username: email,
+    });
+  };
+
+  computeActiveSection = () => {
+    const { purchaseCompleted } = this.state;
+    const { isComplete } = this.props;
+    return purchaseCompleted ? (isComplete ? 3 : 2) : 1;
```

Comment:
Would be good to extract 3/2/1 into constants to represent what they mean. Also `isComplete` is not clear as to what is complete. Possible to make it even more explicit as to what it means.

Reply:
[extracted demoProgressStatus as a enum and renamed isComplete -> isServiceExecutionComplete](https://github.com/singnet/refactored-dapp/pull/127/commits/eaee91c2d143453ad334827d2757d5f4b0fe027f)